GitSearch/Application/Search/SearchController.swift

```diff
@@ -166,19 +147,22 @@ class SearchController: UIViewController, UITextFieldDelegate {
         showHideSpinner(spinner: child)
         
         let keys = keyWords.components(separatedBy: " ")
-//        keys.forEach({key in keys[keys.firstIndex(of: key)!] = "topic:"+key})
         let keysSequence = keys.joined(separator: "+")
-
-        let urlQuery = "https://api.github.com/search/repositories?q="+keysSequence+"&sort=stars&order=desc"
-        fetchRepositoriesHeader(from: urlQuery) { gitRep in
+        
+        SearchServices.shared.getRepositories(keysSequence: keysSequence) { res in
             DispatchQueue.main.async {
                 self.showHideSpinner(spinner: child, false)
-                if gitRep.total_count == 0 {
-                    self.postAlert(title: "No data found", message: "Please, try another  keyword")
-                } else {
-                    let newVC = SearchResultViewController(collectionViewLayout: UICollectionViewFlowLayout())
-                    newVC.gitRep = gitRep
-                    self.navigationController?.pushViewController(newVC, animated: true)
+                switch res {
+                case .success(let gitRep):
+                    if gitRep!.total_count == 0 {
+                        self.postAlert(title: "No data found", message: "Please, try another  keyword")
```

Comment:
When it comes to localization, you are better of with using `NSLocalizedString` everywhere you have user-facing strings. Even if you do not have localization now, its easily possible to generate a list of all UI strings in code later if you've used `NSLocalizedString`. So every programs should wrap any UI text in it from start.